layouts/hub-page.tsx

```diff
@@ -15,9 +15,9 @@ import SkeletonWrapper from "components/atoms/SkeletonLoader/skeleton-wrapper";
 const HubPageLayout = ({ children }: { children: React.ReactNode }) => {
   const router = useRouter();
   const { userId } = useSupabaseAuth();
-  const { filterName } = router.query;
-  const insightId = filterName as string;
-  const { data: insight, isLoading, isError } = useInsight(insightId);
+  const { pageId } = router.query;
+  const insightId = pageId as string;
+  const { data: insight, isLoading, isError } = useInsight(router.query.pageId as string);
```

Comment:
refactor (blocking): `const insightId = pageId as string;` already has the router query param set properly. Use `insightId` instead.

```suggestion
  const { pageId } = router.query;
  const insightId = pageId as string;
  const { data: insight, isLoading, isError } = useInsight(insightId);
```